src/GhostGeneration.jl

```diff
@@ -0,0 +1,128 @@
+    @inbounds for k in 0:(gen.n_ghosts - 1)
+        i = k ÷ gen.resolution
+        # modular division means j cycles through 0-99 100 times (acts like an inner loop)
+        j = k % gen.resolution
+        # index to iterate through the random values
+        index = 2k + 1
+
+        rap = (i + rand_vals[index]) * gen.rapidity_step
+        phi = (j + rand_vals[index + 1]) * gen.phi_step
+
+        # k + 1 is due to indexing beginning at 1 in julia
+        event[n_original + k + 1] = PseudoJet(pt = gen.ghost_pt, rap = rap, phi = phi, _pure_ghost = true)
+    end
+
+    return event
+end
+
+# calculate the number of ghosts in a jet
+function ghosts_in_jet(cluster_seq::ClusterSequence, jet::PseudoJet)
+
+    # Get the constituents of the first jet
+    jet_constituents = JetReconstruction.constituents(jet, cluster_seq)
+
+    # determine the number of ghosts in the jet by checking each constituent's _pure_ghost field
+    num_ghosts = 0
+    for c in jet_constituents
+        if c._pure_ghost
+            num_ghost += 1
+        end
+    end
+
+    # return the number of ghosts
+    return num_ghosts
+end
+
+# calculate the area of a jet
+function GhostedAreaCalculation(cluster_seq::ClusterSequence, jet::PseudoJet, resolution::Int)
+    # the total area is the rapidity range (-5,5) times the phi range (0, 2pi)
+    total_area = 10.0 * 6.28
+
+    gen = GhostedArea(resolution)
+
+    # ghost density is defined as the total number of ghosts divided by the total area
+    ghost_density = gen.n_ghosts / total_area
+
+    # determine the number of ghosts in the jet
+    captured_ghosts = ghosts_in_jet(cluster_seq, jet)
+
+    # area is equal to the number of ghosts in the jet divided by the density of ghosts
+    return captured_ghosts / ghost_density
+end
+
+# returns a vector that contains the number of ghosts in each jet
+function ghosts_in_jets(cluster_seq::ClusterSequence, jets::Vector{PseudoJet})
+
+    # initialize a vector representing the number of ghosts in each jet
+    ghosts_vector = Int[]
+
+    # loop through each jet
+    for jet in jets
+
+        # Get the constituents of the first jet
+        jet_constituents = JetReconstruction.constituents(jet, cluster_seq)
+
+        # determine the number of ghosts in the jet by checking each constituent's _pure_ghost field
+        num_ghosts = 0
+        for c in jet_constituents
+            if c._pure_ghost
+                num_ghost += 1
+            end
+        end
+
+        # add number of ghosts to the vector
+        push!(ghosts_vector, num_ghosts)
+    end
+
+    # return a vector representing the number of ghosts in each jet
+    return ghosts_vector
+end
+
+# returns a vector that contains the calculated area of each jets
+function GhostedAreasCalculation(cluster_seq::ClusterSequence, jets::Vector{PseudoJet}, resolution::Int)
+    # the total area is the rapidity range (-5,5) times the phi range (0, 2pi)
+    total_area = 10.0 * 6.28
+
+    gen = GhostedArea(resolution)
```

Comment:
I don't think this should be called whenever the area of a jet is requested: once the ghosts have been added to the event, this should just need to check `ghosts_in_jets` and divide by the density (to be stored as a member by the constructor).